beacon-chain/forkchoice/types/types.go

```diff
@@ -13,9 +15,17 @@ type ProposerBoostRootArgs struct {
 	SecondsIntoSlot uint64
 }
 
+// Checkpoint is an array version of ethpb.Checkpoint. It is used internally in
+// forkchoice, while the slice version is used in the interface to legagy code
+// in other packages
+type Checkpoint struct {
+	Epoch types.Epoch
+	Root  [fieldparams.RootLength]byte
+}
+
```

Comment:
we need this because in forkchoice all the maps are indexed by `[32]byte`. They can't be slices. Not having this type will make a copy on each function that needs to access forkchoice and this was deemed unacceptable. Once we deprecate store, the whole codepath `blockchain`->`forkchoice` will no longer use `ethpb.Checkpoint`